ROADMAP.md

```diff
@@ -6,21 +6,37 @@ This roadmap covers user-oriented features, UX improvements, JavaScript support,
 
 We hope this updated roadmap provides a clear overview of our plans for k6's future development. As always, we welcome feedback, corrections, and suggestions to make this roadmap more comprehensive, accessible, and valuable for the k6 community.
 
-## Short-term goals
+## Recently completed
```

Comment:
One thing in my mind is to maybe not switch the title from `Short-term goals` to `Recently completed` because we released the modules as `experimental`. Maybe the goal itself is changed a bit. E.g. for the GRPC extension the next milestone is extension stabilization and merging it back to the core. 

Also, we have no listed here, but maybe we should share here some plans about merging to core other experimental extensions that we have, like the xk6-timers and xk6-websockets.

WDYT?

Reply:
>Maybe the goal itself is changed a bit. E.g. for the GRPC extension the next milestone is extension stabilization and merging it back to the core.

>Also, we have no listed here, but maybe we should share here some plans about merging to core other experimental extensions that we have, like the xk6-timers and xk6-websockets.

I would avoid it because I don't think they are short-term so having them make this roadmap unrealistic. Based on our historical data they seem to be mid/long term. I would prefer setting the fact we provided a solution to the main problem as delivered, otherwise, most of the items here will be stuck because they are huge features that never will be implemented in one step (and they could require over years improvements).

Reply:
> I would prefer setting the fact we provided a solution to the main problem as delivered, otherwise, most of the items here will be stuck because they are huge features that never will be implemented in one step (and they could require over years improvements).

Yeah, but that fact is recorded in the release notes, right? The roadmap always will contain something just because it's a roadmap. It's our plan.

Just in case, I'm not saying that these (webcrypto and GRPC) modules should remain in the short-term goals, but I'm trying to say that they should be presented somehow in our roadmap. Otherwise, at some point, they could look abandoned. :cry:

Reply:
I partly agree with Oleg. The roadmap is not the place to list anything that already happened. It's by definition a plan of what will happen in the future. If some goals were completed and shipped, they should just be removed from the roadmap, and new items should take their place.

@olegbespalov They wouldn't look abandoned, just done. As you say, users would be aware of them from our release notes, documentation, etc.

So I vote against changing the title of this section, and for removing the gRPC streaming support and WebCrypto projects. Even if there are still some things to iron out in both, they're probably not worth mentioning here.

Reply:
>They wouldn't look abandoned, just done. As you say, users would be aware of them from our release notes, documentation, etc.

That's the point they aren't just done while they are prefixed as `k6/experimental/`. They are somewhere in their lifecycle (hopefully in the middle), so only if they become just a `k6/` or removed could they be considered done.

Reply:
However, I believe that we should close the #2020 and open a new one about making the `k6/experimental/grpc` stable. And that new task could be presented in a roadmap in mid/long term.

Reply:
I guess this boils down to the "definition of done", but I see it as the feature being ready to use, i.e. generally available. Experimental modules are subject to change, but they're still considered stable and usable.

The support for gRPC streaming and WebCrypto is "done" in that sense, and we can track the minor leftover changes outside of the roadmap. As we discussed on Slack, we don't need to mention minor tasks on the roadmap.

:+1: for closing #2020.

Reply:
>The support for gRPC streaming and WebCrypto is "done" in that sense

I also second this idea. The main issue is to us providing a solution to the problem. The experimental lifecycle should be tracked separately and I see it as a minor issue from the User's point of view (I judge it as minor based on the fact no one opened an issue for asking us to push out a specific module from the experimental phase).